chain/client/src/chunks_delay_tracker.rs

```diff
@@ -10,29 +13,49 @@ use crate::metrics;
 /// Updates the metrics when the head gets updated. Assumes that if the head moves forward, all needed chunks were received.
 /// If a chunk or a block is received or requested multiple times, only the first time is recorded.
 #[derive(Debug, Default)]
-pub(crate) struct ChunksDelayTracker {
+pub struct ChunksDelayTracker {
     /// Timestamps of the first attempt of block processing.
     /// Contains only blocks that are not yet processed.
-    blocks_in_progress: HashMap<CryptoHash, Instant>,
+    pub blocks_in_progress: HashMap<CryptoHash, BlockInProgress>,
     /// An entry gets created when a chunk gets requested for the first time.
     /// Chunks get deleted when the block gets processed.
-    chunks_in_progress: HashMap<ChunkHash, ChunkInProgress>,
+    pub chunks_in_progress: HashMap<ChunkHash, ChunkInProgress>,
+}
+
+#[derive(Debug)]
+pub struct BlockInProgress {
+    /// Timestamp when block was added.
+    pub timestamp: Instant,
+    /// Height at which the block is going to be added.
+    pub height: BlockHeight,
+    /// Hashes of the chunks that belong to this block.
+    pub chunks: Vec<ChunkHash>,
 }
 
 #[derive(Debug)]
 /// Records timestamps of requesting and receiving a chunk. Assumes that each chunk is requested
 /// before it is received.
-struct ChunkInProgress {
+pub struct ChunkInProgress {
     /// Timestamp of requesting a missing chunk.
-    chunk_requested: Instant,
+    pub chunk_requested: Instant,
     /// Timestamp of receiving a chunk.
     /// If a chunk is received multiple times, only the earliest timestamp is recorded.
-    chunk_received: Option<Instant>,
+    pub chunk_received: Option<Instant>,
```

Comment:
actually I think `chunk_received` here actually means the chunk is completed. It means all chunk parts are received, instead of some chunk parts are received. So the `chunks_received` and `chunk_completed` in `UpcomingBlockDebugStatus` are not different, except when some chunks are missing in a block.

Reply:
well spotted. I'll fix in the next PR (probably by adding more state here)